### ISSUE TITLE
memcard-folder-cache: Things still to implement/consider

### ISSUE THREAD

**Issue by ramapcsx2** [PCSX2/pcsx2#666]
Memory cards as folders are supported as of https://github.com/PCSX2/pcsx2/commit/539a1767a3508baeb0348a78fc9a82155dcbd088

There is still some corner cases and some extras that can be done.
- file rename actions get silently dropped. The reason is probably ignored meta data flushes: https://github.com/PCSX2/pcsx2/pull/359#issuecomment-118447045
- currently nothing prevents changes to the memory card folder while emulation is running.
Implement guards against that. One suggestion was keeping an open file handle on the folder content.

Extras:
<ACurtiss> If we want to be super fancy we could additionally add import support for the common file formats you find on GameFAQs and such. Gameshark saves etc.

**Comment by AdmiralCurtiss** [comment]
Also consider:

    Jul 14 19:05:17 <ACurtiss>	Maybe some of the GUI phrasing could be better.
---
    Jul 14 19:08:50 <rama2>	the folder system memory card presents itself as 8mb card?
    Jul 14 19:09:03 <rama2>	my bios browser shows a couple saves, then no more
    Jul 14 19:09:09 <ACurtiss>	Yes, for maximum compatibility.
    Jul 14 19:09:11 <rama2>	the real image (64mb card) has many more games
    Jul 14 19:09:20 <ACurtiss>	Yeah I'm aware of that.
    Jul 14 19:09:38 <ACurtiss>	We could simulate a 64 MB card in the BIOS but 8 MB cards for games, I guess?
    Jul 14 19:13:20 <rama2>	might want to consider the same for homebrew
    Jul 14 19:13:27 <rama2>	if we can somehow reliably detect it

**Comment by AdmiralCurtiss** [comment]
Oh and the underlying cause of #660 probably has a bigger effect on FolderMemoryCards than FileMemoryCards.

**Comment by ramapcsx2** [comment]
Also: Network enabled games will look for any network configuration file on the memory card.
This needs to be considered in the filter.

**Comment by AdmiralCurtiss** [comment]
(BWNETCNF is already included)[https://github.com/PCSX2/pcsx2/commit/e774011d7151507e02aac3425d097758f4440b6e], dunno if there are other network files.

**Comment by ramapcsx2** [comment]
Great :)

**Comment by ramapcsx2** [comment]
So I have a new problem with my 64MB card that I didn't have maybe a week ago. 
Screenshot: https://i.imgur.com/8LaOLYV.png
This is on the same memory card that converted fine when I tested for the initial merge.
Sometime after that, a commit must've broken this.
(error 0: der Vorgang wurde erfolgreich beendet.  >  error 0: success)

**Comment by AdmiralCurtiss** [comment]
Upload the non-converting file somewhere.

**Comment by AdmiralCurtiss** [comment]
Something to consider: It may be worth it to support PSU files (uLaunchELF's save file container, [see here](https://github.com/AKuHAK/uLaunchELF/blob/226df356f1c212591c0a62e8e3cba3a7a2367855/filer.c#L2018) natively, as that solves a bunch of potential metadata and file order related problems at once and makes features such as #1116 significantly easier to implement.

**Comment by eadmaster** [comment]
I second the request to use PSU saves as some japanese games may try to store filenames with unsupported chars.
The [specs for the PSU format are here](https://www.ps2savetools.com/documents/ps2-save-game-format-for-ems-adapter-psu/).

**Comment by AdmiralCurtiss** [comment]
Yeah in retrospect that's definitely the better way to go about it, but I just don't have the time or motivation to implement that.

**Comment by ramapcsx2** [comment]
Then again, file system support for complex character seems to be really good these days.
Might be fine with those Japanese chars :)
